[cyqjoseph] iP

cyqjoseph opened the pull request:
# RoeBot

> I could not come up with a better name

RoeBot frees your mind of having to remember things you need to do. It's
- Easy to use
- Free to use
- Fast to use
All you need to do is,
1. Download it from [here](https://github.com/nus-cs2103-AY2324S2/ip)
2. Double-Click it
3. Add your tasks
4. Let it manage you tasks for you :+1:

And it is once again, **FREE**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

***
If you are a Java Programmer, you can use it to practice Java too. Here's the `code`
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

cyqjoseph committed:
Done Level-0: Rename, Greet, Exit

cyqjoseph committed:
Done Level-1: Echo

cyqjoseph committed:
Done Level-2: Add, List

cyqjoseph committed:
Done Level-3: Mark as Done

cyqjoseph committed:
Done Level-4: ToDo, Event, Deadline

cyqjoseph committed:
Done A-TextUiTesting: Automated Text UI Testing

cyqjoseph committed:
Done Level-5: Handle Errors

cyqjoseph committed:
Done Level-6: Delete

cyqjoseph committed:
Done A-Enums: Use Enums

cyqjoseph committed:
Done Level-7

cyqjoseph committed:
Merge branch-Level-7 into master

cyqjoseph committed:
Done Level-8

cyqjoseph committed:
Merge branch-Level-8 into master

cyqjoseph committed:
Done A-MoreOOP

cyqjoseph committed:
Merge branch-A-MoreOOP into master

cyqjoseph committed:
Done A-Packages

cyqjoseph committed:
Done A-Packages

cyqjoseph committed:
Merge branch-A-Packages into master

cyqjoseph committed:
Merge branch 'add-gradle-support'
Added gradle support

cyqjoseph committed:
Done A-Junit

cyqjoseph committed:
Done A-Jar

cyqjoseph committed:
Done A-JavaDoc

cyqjoseph committed:
Done A-CodingStandard

cyqjoseph committed:
Done Level-9-Find

cyqjoseph committed:
Done A-JavaDoc, A-CodingStandard, Level-9

cyqjoseph committed:
Resolved merge conflicts and merged branch-A-JavaDoc

cyqjoseph committed:
Merged A-CodingStandard branch with master

cyqjoseph committed:
Merge branch 'branch-Level-9'

cyqjoseph committed:
Done A-CheckStyle

cyqjoseph committed:
Add a GUI to the chatbot

Use the JavaFX technology to implement the GUI

Removed the command-line interface for RoeBot application

Removed bugs causing application to not exit

cyqjoseph committed:
Add compiled JAR with bundled JavaFX libraries

Compiled and bundled the application into a standalone JAR file.
Included JavaFX dependencies to ensure compatibility across systems without native JavaFX installation.
Configured Gradle build script (`build.gradle`) to include JavaFX modules in the runtime classpath.
Generated JAR file can now be executed on any system with a compatible Java runtime, independent of JavaFX installation.
Tested the JAR on different environments to ensure functionality.

cyqjoseph committed:
Implement Assertions in Various Classes

This commit introduces the use of Java assertions across multiple classes in the project to validate assumptions about the program's state. Assertions are added to ensure that the program operates under expected conditions, enhancing the robustness and reliability of the code during development.

Changes made:
1. TaskList Class:
        - Added assertions to check if task-related operations are
          performed with valid indices and non-null task objects.
2. Storage Class:
        - Inserted assertions to validate non-null task lists and the
          existence of the file after creation.
3: Parser Class:
        - Added assertions in date parsing methods to ensure non-null
          and properly formatted date strings.
        - Implemented assertions to verify non-empty and non-null user
          input strings.

cyqjoseph committed:
Done A-CodeQuality

Reformatted code to adhere to established coding standards. This
includes consistent use of indentation, bracket and proper import
statements.

Identified and removed redundant code blocks and methods to declutter
codebase and improve overall performance.

Added comprehensie JavaDoc comments where they were lacking. This
includes header comments for modules and detailed descriptions for
complex functions.

cyqjoseph committed:
Done A-Asssertions

In this commit, assertions have been integrated into the project to improve the overall reliability and maintainability of the code. These additions are focused on ensuring that key assumptions in the codebase are explicitly checked during runtime.

cyqjoseph committed:
Merge pull request #1 from cyqjoseph/branch-A-Assertions

Done A-Asssertions

cyqjoseph committed:
Done A-CodeQuality

This commit is focused on improving the code quality of the project by
examining and refactoring the existing code.

Removed unnecessary "System.out.println" statements from code as GUI is
used over CLI.

cyqjoseph committed:
Merge pull request #2 from cyqjoseph/branch-A-CodeQuality

Done A-CodeQuality

miffi commented on src/main/java/commands/AddEventCommand.java:
```diff
@@ -0,0 +1,35 @@
+
+    @Override
+    public UserCommand execute(TaskList taskList, UI ui, Storage storage) {
+        Event event = new Event(this.name, this.start, this.end);
```
`this.name`, `this.start`, and `this.end` don't need the `this` as there are no conflicting names in the local scope. According to the standard, it's better to leave the `this` out in this case, so  I think removing the `this` here and in some other parts of the code I'm seeing will be cleaner.

miffi commented on src/main/java/commands/DeleteTaskCommand.java:
```diff
@@ -0,0 +1,30 @@
+/**
+ * Deletes a task from the task list.
+ */
+
```
The empty line between class and Javadoc header should be removed.

miffi commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+        this.by = by;
+    }
+
+    public String formatBy() {
```
This method needs a javadoc header. I can't really understand what the function does just by looking at the name.

miffi commented on src/main/java/commands/AbstractCommand.java:
```diff
@@ -0,0 +1,12 @@
+ * Represents an executable command as an abstract class.
+ */
+public abstract class AbstractCommand {
+    public abstract UserCommand execute(TaskList taskList, UI ui, Storage storage);
```
Non-private method needs a javadoc header.

miffi commented on src/main/java/roebot/RoeBot.java:
```diff
@@ -0,0 +1,47 @@
+        try {
+            storage.loadTasks(this.taskList);
+        } catch (IOException e) {
+            System.out.println("FIX THIS ERROR");
```
This got a chuckle out of me haha

miffi commented on src/main/java/services/Storage.java:
```diff
@@ -0,0 +1,69 @@
+            if (!file.exists()) {
+                file.getParentFile().mkdirs();
+                file.createNewFile();
+                assert file.exists() : "File should exist after creation";
```
Nice use of asserts for checking invariants. Completely skipped my mind that they existed. I should use them as well.

cyqjoseph replied:
Thanks for the comments!

miffi commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,40 @@
+        this.isDone = false;
+    }
+
+    public String getName() {
```
Missing JavaDoc header

miffi reviewed (commented):
Hello :). I'm a peer reviewer for your iP this week. This is my first review (the coding standard one).

As far as I can see, you have followed the coding standard really closely, so I don't have much to comment. I have just pointed out some nits (and some general comments), but otherwise LGTM!

cyqjoseph committed:
Done C-DetectDuplicates

This commit introduces a significant enhancement to the project to
filter for duplicates.

Added isDuplicateTask method in TaskList.
   - For Deadline tasks, it checks for duplicates based on both name and deadline time.
   - For Event tasks, it verifies if either the start or end time, along with the task name, is identical to an existing task.
   - For other task types, a simple name-based comparison is performed.

Modified the addTask method to prevent addition of duplicate tasks.

cyqjoseph committed:
Done A-UserGuide

Add comprehensive user guide documentation

- Include step-by-step instructions for users
- Provide descriptions of application features
- Add screenshots for visual guidancee

cyqjoseph committed:
Merge pull request #3 from cyqjoseph/branch-A-UserGuide

Done A-UserGuide

cyqjoseph committed:
Fix Ui.png name

cyqjoseph committed:
Truncate unnecessarily long methods

cyqjoseph committed:
Modify archiveBaseName in shadowJar in build.gradle config file.

cyqjoseph committed:
Rename archiveBaseName to roebot after naming issues post-build phase

FAT Jar file was named roebot.jar.jar after build so archiveBaseName was
renamed to just "roebot".

cyqjoseph committed:
Add getter methods for start and end properties.

This commit introduces two getter methods in the class:

getStart(): Returns the 'start" property of type LocalDateTime
getEnd(): Returns the 'end' property of type LocalDateTime

cyqjoseph committed:
Add getter methods for name and status properties

This commit introduces two getter methods in Task class:

getName(): Returns the 'name' property of type String.
getStatus(): Returns the 'status' property of type Boolean.

cyqjoseph committed:
Add JavaDoc for formatBy method in Deadline.

This commit introduces a JavaDoc comment for a single method in
Deadline.

formatBy require clearer description as to what the method does.

cyqjoseph committed:
Add JavaDoc comments for markAsDone and markAsUndone.

This commit introduces two JavaDoc comments in the class Task

1. markAsDone(): Marks task as Done
2. markAsUndone(): Marks task as Undone.

cyqjoseph committed:
Change file path in build.gradle.

Previously mainClass was set to Main, which was incorrect, fix to
Launcher class required to allow build jar file to run properly.

Update new jar file on github release tab.

cyqjoseph committed:
Fix ArrayIndexOutOfBoundsException in parseTaskFromString

This commit addresses an issue where `parseTaskFromString` could throw an ArrayIndexOutOfBoundsException if the input string was not properly formatted. Replaced assert statements with explicit exception handling to ensure robust error checking, even when assertions are disabled at runtime.

Additionally, updated build.gradle to include platform-specific JavaFX dependencies for Windows, macOS, and Linux to support cross-platform functionality.

Dependencies added for javafx-base, javafx-controls, javafx-fxml, and javafx-graphics for each respective platform using the javaFxVersion variable for consistent version management.
